### ISSUE TITLE
ESLint is taking too long to run --max-warnings doesn't work

### ISSUE THREAD

**Issue by fab1o** [eslint/eslint#12936]
We have over 40,000 eslint issues (between warnings and errors) in our codebase and running ESLint 6.8.0 takes a long time to finish.

`--max-warnings` which teorically should exit after an x number of issues was found, doesn't work.

Is the command wrong?

```sh
eslint . --ext .js --max-warnings 3
```

**Comment by kaicataldo** [comment]
`--max-warnings` is designed to set a threshold for treating rules that are set to report as `"warning"` (not `"error"`) to change the exit code value when ESLint completes linting, so it's going to help with this case.

I would recommend running the command on smaller groupings of files.

**Comment by maddhruv** [comment]
@fab1o I hope you have probably added `.eslintignore` and not running on `node_modules| as well.

Moreover, can you share your eslint config and screenshot of errors

**Comment by fab1o** [comment]
Yes, `node_modules` is ignored by `.eslintignore`.

@kaicataldo Gotcha. So I completely misunderstood this option.

My goal is to have eslint exit after a certain amount of warnings is found thus giving me the prompt back faster. So after finding 3 warnings, it would exit and not continue linting. Is there an option for that?

**Comment by kaicataldo** [comment]
We don't have an option for that, no. If you'd like to create an [RFC](https://github.com/eslint/rfcs) for an option that makes ESLint bail after `x` number of errors are found, please feel free!
